### PR TITLE
Wallet identify GRPC

### PR DESCRIPTION
**Pull request by StriderDM**
## Description
This PR adds identify to the GRPC methods of the wallet to be able to
query out the public key and public address for the wallet as those
fields now exist within the database instead of a standalone file.

Needs https://github.com/tari-project/wallet-grpc-client/pull/7 to be merged first since the dependency of the `package.json` in the integration test is currently pointing to my fork of it.

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [ ] I'm merging against the `development` branch.
* [ ] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: integration_tests/features/support/steps.js [hunk 1]
```diff
@@ -268,26 +275,26 @@ When(/I send (.*) tari from (.*) to (.*),(.*) at fee (.*)/, async function (tari
...
 });
 
 Then(/Batch transfer of (.*) transactions was a success from (.*) to (.*),(.*)/,  async function (txCount,walletA,walletB,walletC) {
-   var WalletA = this.getWallet(walletA);
-   var ClientA = WalletA.getClient();
-   var WalletB = this.getWallet(walletB);
-   var ClientB = WalletB.getClient();
-   var WalletC = this.getWallet(walletC);
-   var ClientC = WalletC.getClient();
+   let WalletA = this.getWallet(walletA);
```
Comment (stringhandler) [inline integration_tests/features/support/steps.js]: JS vars should be camelcase

```suggestion
   let walletA = this.getWallet(walletA);
```

File: applications/tari_app_grpc/proto/wallet.proto [hunk 2]
```diff
@@ -39,6 +42,13 @@ service Wallet {
 
 message GetVersionRequest { }
 
+message GetIdentityRequest { }
+
+message GetIdentityResponse {
+    string public_key = 1;
```
Comment (stringhandler) [inline applications/tari_app_grpc/proto/wallet.proto]:
```suggestion
    bytes public_key = 1;
```

**Review by stringhandler — COMMENTED** [review]
Looks good on the whole, just some comments

File: integration_tests/helpers/config.js [hunk 3]
```diff
@@ -60,21 +59,23 @@ function baseEnvs(peerSeeds = [])
      return envs;
 }
 
-function createEnv(nodeFile="newnodeid.json",walletGrpcAddress, walletGrpcPort, baseNodeGrpcAddress="127.0.0.1",baseNodeGrpcPort="8080", baseNodePort="8080",proxyFullAddress="127.0.0.1:8080",options, peerSeeds=[]) {
+function createEnv(name="config_identity",isWallet=false, nodeFile="newnodeid.json",walletGrpcAddress="127.0.0.1", walletGrpcPort="8082", walletPort="8083", baseNodeGrpcAddress="127.0.0.1", baseNodeGrpcPort="8080", baseNodePort="8081",proxyFullAddress="127.0.0.1:8084",options, peerSeeds=[]) {
```
Comment (hansieodendaal) [inline integration_tests/helpers/config.js:62]: nit: Code can be simplified.

`isWallet=false,` is not needed here, whereby `walletPort="8083",` and `baseNodePort="8081",` can be combined into one variable called `nodePort="8081",` or similar.

File: integration_tests/helpers/baseNodeProcess.js [hunk 4]
```diff
@@ -90,7 +90,7 @@ class BaseNodeProcess {
                 fs.mkdirSync(this.baseDir + "/log", {recursive: true});
             }
 
-            let envs = createEnv(this.nodeFile,"127.0.0.1", "8080","127.0.0.1",this.grpcPort,this.port,"127.0.0.1:8080",this.options,this.peerSeeds);
+            let envs = createEnv(this.name,false, this.nodeFile,"127.0.0.1", "8082","8081","127.0.0.1",this.grpcPort,this.port,"127.0.0.1:8080",this.options,this.peerSeeds);
```
Comment (hansieodendaal) [inline integration_tests/helpers/baseNodeProcess.js:93]: nit: Code can be simplified.

Suggest only pass non-default values as named parameters.

File: integration_tests/helpers/mergeMiningProxyProcess.js [hunk 5]
```diff
@@ -30,7 +30,7 @@ class MergeMiningProxyProcess {
             }
 
             let proxyAddress = "127.0.0.1:"+this.port;
-            let envs = createEnv("nodeid.json",this.walletAddress,this.walletGrpcPort,this.nodeAddress,this.nodeGrpcPort, this.baseNodePort, proxyAddress,[], []);
+            let envs = createEnv(this.name,false, "nodeid.json",this.walletAddress,this.walletGrpcPort,this.port,this.nodeAddress,this.nodeGrpcPort, this.baseNodePort, proxyAddress,[], []);
```
Comment (hansieodendaal) [inline integration_tests/helpers/mergeMiningProxyProcess.js:33]: nit: Code can be simplified.

Suggest only pass non-default values as named parameters.

File: integration_tests/helpers/walletProcess.js [hunk 6]
```diff
@@ -57,7 +49,7 @@ class WalletProcess {
                 fs.mkdirSync(this.baseDir + "/log", {recursive: true});
             }
 
-           let envs = createEnv("nodeid.json","127.0.0.1",this.grpcPort, this.baseNodeGrpcAddress, this.baseNodeGRPCPort,this.baseNodePort, "127.0.0.1:8080", [], this.peerSeeds)
+           let envs = createEnv(this.name,true, "nodeid.json","127.0.0.1",this.grpcPort, this.port, this.baseNodeGrpcAddress, this.baseNodeGRPCPort,this.baseNodePort, "127.0.0.1:8081", [], this.peerSeeds)
```
Comment (hansieodendaal) [inline integration_tests/helpers/walletProcess.js:52]: nit: Code can be simplified.

Suggest only pass non-default values as named parameters.

File: integration_tests/helpers/walletProcess.js [hunk 7]
```diff
@@ -10,6 +10,7 @@ class WalletProcess {
 
     constructor(name, baseNodeAddress, baseNodePort, baseNodeGrpcPort) {
             this.name = name;
+            this.nodeFile = "cwalletid.json";
             this.baseNodeAddress = baseNodeAddress;
             this.baseNodePort = baseNodePort;
             this.baseNodeGrpcPort = baseNodeGrpcPort;
```
Comment (hansieodendaal) [inline integration_tests/helpers/walletProcess.js:16]: nit: Reference to base node values are confusing in the wallet and should be removed

File: integration_tests/helpers/walletProcess.js [hunk 8]
```diff
@@ -10,6 +10,7 @@ class WalletProcess {
 
     constructor(name, baseNodeAddress, baseNodePort, baseNodeGrpcPort) {
```
Comment (hansieodendaal) [inline integration_tests/helpers/walletProcess.js:11]: nit: `baseNodeAddress, baseNodePort, baseNodeGrpcPort` are not needed here

File: integration_tests/helpers/config.js [hunk 9]
```diff
@@ -60,21 +59,23 @@ function baseEnvs(peerSeeds = [])
...
              TARI_BASE_NODE__LOCALNET__BASE_NODE_IDENTITY_FILE: `${nodeFile}`,
-             TARI_BASE_NODE__LOCALNET__TCP_LISTENER_ADDRESS: "/ip4/0.0.0.0/tcp/" + `${baseNodePort}`,
-             TARI_BASE_NODE__LOCALNET__PUBLIC_ADDRESS: "/ip4/127.0.0.1/tcp/" + `${baseNodePort}`,
+             TARI_BASE_NODE__LOCALNET__TCP_LISTENER_ADDRESS: "/ip4/0.0.0.0/tcp/" + (isWallet ? `${walletPort}` : `${baseNodePort}`),
+             TARI_BASE_NODE__LOCALNET__PUBLIC_ADDRESS: "/ip4/127.0.0.1/tcp/" + (isWallet ? `${walletPort}` : `${baseNodePort}`),
              TARI_MERGE_MINING_PROXY__LOCALNET__PROXY_HOST_ADDRESS: `${proxyFullAddress}`,
+             TARI_BASE_NODE__LOCALNET__TRANSPORT: "tcp",
          }
+         console.log(name);
          console.log(configEnvs);
```
Comment (hansieodendaal) [inline integration_tests/helpers/config.js:74]: nit: Probably these prints can be commented out as a default and uncommented if someone needs debug info

File: integration_tests/helpers/mergeMiningProxyClient.js [hunk 10]
```diff
@@ -23,7 +23,7 @@ class MergeMiningProxyClient {
                 "reserve_size": 60
             }
         });
-        console.log(res.data);
+        console.log("Blocktemplate:",res.data.result.blocktemplate_blob);
```
Comment (hansieodendaal) [inline integration_tests/helpers/mergeMiningProxyClient.js:26]: nit: Probably this print can be commented out as a default and uncommented if someone needs debug info

File: integration_tests/helpers/walletProcess.js [hunk 11]
```diff
@@ -25,18 +26,9 @@ class WalletProcess {
         this.port = await getFreePort(19000, 25000);
```
Comment (hansieodendaal) [inline integration_tests/helpers/walletProcess.js:26]: nit: Code can be simplified.

This function is not needed:
```
    async preInit() {
        this.baseNodeAddress = "127.0.0.1";
        this.baseNodePort = await getFreePort(19000, 25000);
        this.baseNodeGrpcPort = await getFreePort(19000, 25000);
    }
```

File: applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs [hunk 12]
```diff
@@ -50,6 +52,16 @@ impl wallet_server::Wallet for WalletGrpcServer {
...
     }
 
+    async fn identify(&self, request: Request<GetIdentityRequest>) -> Result<Response<GetIdentityResponse>, Status> {
+        let _request = request.into_inner();
+
+        let node_id = self.wallet.comms.node_identity();
+        Ok(Response::new(GetIdentityResponse {
+            public_key: node_id.public_key().to_string(),
+            public_address: node_id.public_address().to_string(),
+        }))
```
Comment (hansieodendaal) [inline applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs:63]: Suggest add `node_id.node_id().to_string()` as well

File: applications/tari_app_grpc/proto/wallet.proto [hunk 13]
```diff
@@ -39,6 +42,13 @@ service Wallet {
...
+    string public_address = 2;
+}
```
Comment (hansieodendaal) [inline applications/tari_app_grpc/proto/wallet.proto:51]: Suggest add `string node_id = 3;` as well

**Review by hansieodendaal — COMMENTED** [review]
Nice additional interface here. Please see comments below.

File: integration_tests/features/support/steps.js (hunk 1 above)
Comment (StriderDM) [reply integration_tests/features/support/steps.js]: Think I got all of them

File: applications/tari_app_grpc/proto/wallet.proto (hunk 2 above)
Comment (StriderDM) [reply applications/tari_app_grpc/proto/wallet.proto]: Done, added node_id of type bytes as well.

File: applications/tari_app_grpc/proto/wallet.proto (hunk 13 above)
Comment (StriderDM) [reply applications/tari_app_grpc/proto/wallet.proto:51]: Done

File: applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs (hunk 12 above)
Comment (StriderDM) [reply applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs:63]: Done, was sent at bytes, same as public_key now

**Review by hansieodendaal — APPROVED** [review]
LGTM

**Pull request merged**